controllers/release/release_adapter_test.go

```diff
@@ -0,0 +1,414 @@
+		Expect(err).Should(HaveOccurred())
+	})
+
+	It("can return the pipelineRun from the release within the give adapter", func() {
+		Eventually(func() bool {
+			result, err := cacheAdapter.EnsureReleasePipelineRunExists()
+			return (!result.CancelRequest && err == nil)
+		}).Should(BeTrue())
+		pipelineRun, err := cacheAdapter.getReleasePipelineRun()
+		Expect(err).Should(Succeed())
+		Expect(reflect.TypeOf(pipelineRun)).
+			To(Equal(reflect.TypeOf(&tektonv1beta1.PipelineRun{})))
+	})
+
+	It("can return the releaseStrategy from a given releaseLink to the given adapter", func() {
+		strategy, err := adapter.getReleaseStrategy(releaseLink)
+		Expect(err).Should(Succeed())
+		Expect(reflect.TypeOf(strategy)).
+			To(Equal(reflect.TypeOf(&appstudiov1alpha1.ReleaseStrategy{})))
+	})
+
+	It("can return the releaseStrategy from a given adapter", func() {
+		strategy, _ := adapter.getReleaseStrategyFromRelease()
+		Expect(reflect.TypeOf(strategy)).
+			To(Equal(reflect.TypeOf(&appstudiov1alpha1.ReleaseStrategy{})))
+	})
+
+	It("can return the target ReleaseLink from a given ReleaseLink within a given adapter", func() {
+		link, _ := adapter.getTargetReleaseLink()
+		Expect(reflect.TypeOf(link)).
+			To(Equal(reflect.TypeOf(&appstudiov1alpha1.ReleaseLink{})))
+	})
+
+	It("can mark the status for a given Release according to the pipelineRun status", func() {
+		release.MarkRunning()
+		pipelineRun := tekton.NewReleasePipelineRun("test-pipeline", testNamespace).AsPipelineRun()
+
+		pipelineRun.Status.MarkSucceeded("suceedeed", "set it to succeeded")
+		err := adapter.registerReleasePipelineRunStatus(pipelineRun)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(release.HasSucceeded()).To(BeTrue())
+
+		release.Status.Conditions = []metav1.Condition{} // Clear up previous condition so isDone returns false
+		pipelineRun.Status.MarkFailed("failed", "set it to failed")
+		err = adapter.registerReleasePipelineRunStatus(pipelineRun)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(release.HasSucceeded()).To(BeFalse())
+	})
+
+	It("can add the release information to its status and mark it as Running", func() {
+		// err should be nil if releaseStrategy or pipelineRun is nil
+		err := adapter.registerReleaseStatusData(nil, nil)
+		Expect(err).To(BeNil())
+
+		pipelineRun := tekton.NewReleasePipelineRun("test-pipeline", testNamespace).AsPipelineRun()
+		pipelineRun.Name = "new-name"
+		err = adapter.registerReleaseStatusData(pipelineRun, releaseStrategy)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(release.Status.ReleasePipelineRun, pipelineRun.GetName())
+		Expect(release.Status.Conditions[0].Status == "Unknown")
+		Expect(release.Status.Conditions[0].Reason == "Running")
+	})
+
+	It("can update the Status of a release within the given adapter", func() {
+		applicationSnapshot.TypeMeta.Kind = "ApplicationSnapshot"
+
+		pipelineRun := tekton.NewReleasePipelineRun("test-pipeline", testNamespace).AsPipelineRun()
+		pipelineRun.SetName("test")
```

Comment:
In a previous test you use `pipelineRun.Name`. Choose one so we can be consistent ;)